src/Security/Authentication/Certificate/src/CertificateAuthenticationOptions.cs

```diff
@@ -23,18 +23,30 @@ public class CertificateAuthenticationOptions : AuthenticationSchemeOptions
     /// </summary>
     public X509Certificate2Collection CustomTrustStore { get; set; } = new X509Certificate2Collection();
 
+    /// <summary>
+    /// A callback that will be invoked to dynamically select the trusted X509 certificate components of the certificate chain.
+    /// This is higher priority than <see cref="CustomTrustStore"/>.
+    /// </summary>
+    public Func<X509Certificate2, X509Certificate2Collection>? CustomTrustStoreSelector { get; set; }
```

Comment:
Should the HttpContext (or some other object) be passed in for context?

Reply:
Move these to https://github.com/dotnet/aspnetcore/blob/52d1c3020ac9b0f747b3ff0c244e7ec2be6f5554/src/Security/Authentication/Certificate/src/Events/CertificateAuthenticationEvents.cs and follow the pattern there?

Reply:
I think this is a good idea! My latest iteration now leverages these events.